lib/sensors/MoistureSensor.cpp

```diff
@@ -0,0 +1,21 @@
+#include "mbed.h"
+#include <I2C.h>
+#include "MoistureSensor.h"
+
+MoistureSensor::MoistureSensor(PinName sda, PinName scl, char* data, int address){
+    I2C i2c(sda, scl);    
```

Comment:
this object goes out of scope right after this constructor. did u mean to have a member variable declared in the class and intialize it here?